src/commands.rs

```diff
@@ -332,33 +332,40 @@ impl CommandManager {
     }
 
     fn handle_callbacks(&self, s: &mut Cursive, cmd: &Command) -> Result<Option<String>, String> {
-        let local = if let Some(mut contextmenu) = s.find_name::<ContextMenu>("contextmenu") {
-            contextmenu.on_command(s, cmd)?
-        } else if let Some(mut add_track_menu) = s.find_name::<AddToPlaylistMenu>("addtrackmenu") {
-            add_track_menu.on_command(s, cmd)?
-        } else if let Some(mut select_artist) = s.find_name::<SelectArtistMenu>("selectartist") {
-            select_artist.on_command(s, cmd)?
-        } else if let Some(mut select_artist_action) =
-            s.find_name::<SelectArtistActionMenu>("selectartistaction")
-        {
-            select_artist_action.on_command(s, cmd)?
-        } else {
-            s.on_layout(|siv, mut l| l.on_command(siv, cmd))?
+        let local = match s.find_name::<ContextMenu>("contextmenu") {
+            Some(mut contextmenu) => contextmenu.on_command(s, cmd)?,
+            _ => match s.find_name::<AddToPlaylistMenu>("addtrackmenu") {
+                Some(mut add_track_menu) => add_track_menu.on_command(s, cmd)?,
+                _ => match s.find_name::<SelectArtistMenu>("selectartist") {
+                    Some(mut select_artist) => select_artist.on_command(s, cmd)?,
+                    _ => match s.find_name::<SelectArtistActionMenu>("selectartistaction") {
+                        Some(mut select_artist_action) => {
+                            select_artist_action.on_command(s, cmd)?
+                        }
+                        _ => s.on_layout(|siv, mut l| l.on_command(siv, cmd))?,
+                    },
+                },
+            },
         };
 
-        if let CommandResult::Consumed(output) = local {
-            Ok(output)
-        } else if let CommandResult::Modal(modal) = local {
-            s.add_layer(modal);
-            Ok(None)
-        } else if let CommandResult::View(view) = local {
-            s.call_on_name("main", move |v: &mut Layout| {
-                v.push_view(view);
-            });
-
-            Ok(None)
-        } else {
-            self.handle_default_commands(s, cmd)
+        match local {
+            CommandResult::Consumed(output) => Ok(output),
+            _ => match local {
+                CommandResult::Modal(modal) => {
+                    s.add_layer(modal);
+                    Ok(None)
+                }
+                _ => match local {
+                    CommandResult::View(view) => {
+                        s.call_on_name("main", move |v: &mut Layout| {
+                            v.push_view(view);
+                        });
+
+                        Ok(None)
+                    }
+                    _ => self.handle_default_commands(s, cmd),
+                },
+            },
```

Comment:
Was this suggested by the toolchain? I find the nested matches harder to read than the `if let` expressions.

Reply:
Yeah, I also find it way harder to read. There were even places with like 8 closing curly brackets to close all the nested scopes. This was automatically done by Cargo when I executed `cargo fix --edition`. I have no clue why.

The only manual changes I made in this PR were changes to the workspace definitions in the two `Cargo.toml` files to make future edition updates easier. I think those are in the second commit where I also increased the edition number from 2021 to 2024.